src/main/java/com/exasol/sql/dml/merge/Merge.java

```diff
@@ -71,42 +89,60 @@ public Merge on(final BooleanExpression condition) {
         return this;
     }
 
+    /**
+     * Get the merge condition.
+     *
+     * @return criteria that must be met for the rows in source and destination to be considered a match.
+     */
+    public BooleanExpression getCondition() {
+        return this.condition;
```

Comment:
Seems like this `condition` should returned from `on` object.

Reply:
Fixed. Also fixed names.